packages/core/src/modules/dids/DidsApi.ts

```diff
@@ -99,4 +101,63 @@ export class DidsApi {
   public getCreatedDids({ method, did }: { method?: string; did?: string } = {}) {
     return this.didRepository.getCreatedDids(this.agentContext, { method, did })
   }
+
+  /**
+   * Import an existing did that was created outside of the DidsApi. This will create a `DidRecord` for the did
+   * and will allow the did to be used in other parts of the agent. If you need to create a new did document,
+   * you can use the {@link DidsApi.create} method to create and register the did.
+   *
+   * If no `didDocument` is provided, the did document will be resolved using the did resolver. You can optionally provide a list
+   * of private key buffer with the respective private key bytes. These keys will be stored in the wallet, and allows you to use the
+   * did for other operations.
+   *
+   * By default, this method will throw an error if the did already exists in the wallet. You can override this behavior by setting
+   * the `overwrite` option to `true`. This will update the did document in the record, and allows you to update the did over time.
+   */
+  public async import({ did, didDocument, privateKeys = [], overwrite }: ImportDidOptions) {
+    if (didDocument && didDocument.id !== did) {
+      throw new AriesFrameworkError(`Did document id ${didDocument.id} does not match did ${did}`)
+    }
+
+    const existingDidRecord = await this.didRepository.findCreatedDid(this.agentContext, did)
+    if (existingDidRecord && !overwrite) {
+      throw new AriesFrameworkError(
+        `A created did ${did} already exists. If you want to override the existing did, set the 'overwrite' option to update the did.`
+      )
+    }
+
+    if (!didDocument) {
+      didDocument = await this.resolveDidDocument(did)
+    }
+
+    // Loop over all private keys and store them in the wallet. We don't check whether the keys are actually associated
+    // with the did document, this is up to the user.
+    for (const key of privateKeys) {
+      // TODO: Should we check if the keys are already in the wallet?
```

Comment:
If a key is already in the wallet, Wallet object is supposed to make us know about it by throwing an Error. The problem is: what's the resulting wallet state if only some of the provided keys are already present? If we got an error when calling this method because of this, we cannot tell afterwards which ones were actually saved, so it could be confusing.

It could be also confusing the fact that probably we set the `overwrite` flag and we could expect not only the DidRecord to be overwritten if already existing (because we are an average developer that does not read the JSDoc :stuck_out_tongue: ).

I think it will be safer to check all provided keys for presence before doing anything that can modify wallet state. But of course this could be part of another PR, once we update Wallet API appropiately.

Reply:
Good points. Let me see how easy it is to check if a key exists. I think I'll have to do some updates to the indy-sdk-react-native for this. Oh well  🤷

Reply:
Ok, so there's no way to check if a key exists in indy-sdk. You can get metadata, but this will only return something if there's actual metadta for that key.

It does throw `WalletItemAlreadyExists` error, which we could check if a key already exists by actually thying to create it. But yeah, then you would have the key in the wallet.

So what I'm going to pursue now (please mention if you have a better approach), is the following: loop through all private keys and call createKey. If an error is thrown that the key already exists we just continue as the intended outcome is the same: all private keys are in the wallet.

Reply:
Agree, I think that's the best we can do for the time being. Would be nice to quickly explain in a comment or add a TODO to remember about it and improve when indy-sdk finally go to a better life ;-).